src/pages/index.js

```diff
@@ -18,42 +19,201 @@ import {
+  .then((userData) => {
+    userInfo.setUserInfo({
+      name: userData.name,
+      description: userData.about,
+      avatar: userData.avatar,
+    });
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
 // User Info
-const userInfo = new UserInfo(".profile__title", ".profile__description");
+const userInfo = new UserInfo({
+  nameSelector: ".profile__title",
+  descriptionSelector: ".profile__description",
+  avatarSelector: ".profile__image",
+});
+
+// Function to handle click event
+function handleImageClick({ name, link }) {
+  popupWithImage.open(name, link);
+}
+
+function createCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteCard,
+    handleLike
+  );
+  return card.getView();
+}
+
+function handleProfileEditSubmit({ name, description }) {
+  profileEditPopup.setLoading(true);
+  api
+    .updateUserInfo(name, description)
+    .then(() => {
+      userInfo.setUserInfo({ name: name, description: description });
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileEditPopup.setLoading(false);
+      profileEditPopup.close();
+    });
+}
+
+function handleAddImageSubmit(name, url) {
+  addCardPopup.setLoading(true);
+  api
+    .addCard(name, url)
+    .then((cardData) => {
+      const card = createCard(cardData);
+      section.addItem(card);
+      addCardPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      addCardPopup.setLoading(false);
+    });
+}
+
+function handleAvatarSubmit(url) {
+  profileAvatarPopup.setLoading(true);
+  api
+    .updateAvatar(url)
+    .then((userData) => {
+      userInfo.setAvatar(userData);
+      profileAvatarPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileAvatarPopup.setLoading(false);
+    });
+}
+
+function handleDeleteCard(cardId) {
+  cardDeletePopup.open();
+  cardDeletePopup.setSubmitAction(() => {
+    cardDeletePopup.setLoading(true, "Deleting...");
+    api
+      .deleteCard(cardId)
+      .then(() => {
+        this.handleDeleteCard();
+        cardDeletePopup.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        cardDeletePopup.setLoading(false, "Yes");
+      });
+  });
+}
+
+function handleLike(cardId) {
+  if (cardId._isLiked) {
+    api
+      .dislikeCard(cardId._id)
+      .then(() => {
+        cardId.handleLikeIcon();
+        cardId._isLiked = false;
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+  if (!cardId._isLiked) {
+    api
+      .likeCard(cardId._id)
+      .then(() => {
+        cardId.handleLikeIcon();
+        cardId._isLiked = true;
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+}
+// Form Validator
+
+const avatarFormValidator = new FormValidator(config, profileAvatarForm);
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object `formValidators`.  And then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```